### PR TITLE
Feature: Eslint i18n validation

### LINKED ISSUE #374: 🚀 [Feature] Ensure completeness of a language
## 🚀 Feature

Ensure completeness of a language.

- Make sure that all language keys are defined for a specific language compared to the English translation.
- Remove all languages not complete from the release
- Have required languages to allow a release (e.g. German, English, French)
- Use Github workflows to make those checks

### LINKED ISSUE #1573: refactor follow steps
Follow ups:
 * Create a component for the collapse icon https://github.com/gradido/gradido/issues/1599
 * think of a component for the collapse itself https://github.com/gradido/gradido/issues/1599
 * find a better way to handle the minus sign https://github.com/gradido/gradido/issues/1575
 * think of vue filter for GDD in general
 * do we really want to have `Uhr` after dates displayed?

_Originally posted by @Mogge in https://github.com/gradido/gradido/pull/1548#pullrequestreview-901845843_

### LINKED ISSUE #1614: 🔧 [Refactor] yarn i18n:report
make a 
- yarn i18n:report

remove all unused lines

### PR DESCRIPTION
**Pull request by ulfgebhardt**
## 🍰 Pullrequest

Validate our language files with eslint:
- find missing keys
- find unused keys
- no more stray text in the application
- more rules for i18n

### Issues

- fixes #1614
- fixes #374
- fixes #1573

### Todo

- [x] Confirm all required keys are there (visual inspection was good)
- [x] Remove `i18n:report` and corresponding plugins if possible
- [x] Admin
- [x] check the dynamic keys again
- [x] fix tests

### CODE REVIEW

File: frontend/src/locales/de.json [hunk 1]
```diff
@@ -130,19 +117,29 @@
...
+    "div": "/",
+    "equal": "=",
+    "exclaim": "!",
+    "pipe": "|"
+  },
+  "navigation": {
+    "admin_area": "Adminbereich",
+    "logout": "Abmelden",
+    "members_area": "Mitgliederbereich",
+    "my-profil": "Mein Profil",
```
Comment (Mogge) [inline frontend/src/locales/de.json:133]:
```suggestion
    "profile": "Mein Profil",
```

File: frontend/src/locales/en.json [hunk 2]
```diff
@@ -130,19 +117,29 @@
...
+    "div": "/",
+    "equal": "=",
+    "exclaim": "!",
+    "pipe": "|"
+  },
+  "navigation": {
+    "admin_area": "Admin Area",
+    "logout": "Logout",
+    "members_area": "Members area",
+    "my-profil": "My profile",
```
Comment (Mogge) [inline frontend/src/locales/en.json:133]:
```suggestion
    "profile": "My profile",
```

File: frontend/src/locales/en.json [hunk 3]
```diff
@@ -169,40 +165,15 @@
...
-      "reset-password": {
-        "text": "Enter a new password that you will use to log in to your Gradido account in the future.."
-      },
-      "send_now": "Send now",
-      "set": "Set password",
-      "set-password": {
-        "text": "Now save your new password, which you can use to log in to your Gradido account in the future."
-      },
-      "subtitle": "If you have forgotten your password, you can reset it here."
+      "forgot_pwd": "Forgot password?"
```
Comment (Mogge) [inline frontend/src/locales/en.json:168]: They are used in ForgotPassword.vue

File: frontend/src/pages/ForgotPassword.vue [hunk 4]
```diff
@@ -5,7 +5,9 @@
         <div class="header-body text-center mb-7">
           <b-row class="justify-content-center">
             <b-col xl="5" lg="6" md="8" class="px-2">
+              <!-- eslint-disable-next-line @intlify/vue-i18n/no-dynamic-keys-->
               <h1>{{ $t(displaySetup.headline) }}</h1>
+              <!-- eslint-disable-next-line @intlify/vue-i18n/no-dynamic-keys-->
```
Comment (Mogge) [inline frontend/src/pages/ForgotPassword.vue:10]: You deleted those keys. I suggest to grep this lint instruction to find more keys that are generated by code.

File: frontend/src/pages/Register.vue [hunk 5]
```diff
@@ -100,6 +100,7 @@
                         v-model="form.agree"
                         :name="$t('site.signup.agree')"
                       >
+                        <!-- eslint-disable-next-line @intlify/vue-i18n/no-v-html -->
                         <span class="text-muted" v-html="$t('site.signup.agree')"></span>
```
Comment (Mogge) [inline frontend/src/pages/Register.vue:104]:
```suggestion
                        <span class="text-muted">{{ $t('site.signup.agree') }}</span>
```

File: frontend/src/pages/Register.vue [hunk 6]
```diff
@@ -100,6 +100,7 @@
                         v-model="form.agree"
                         :name="$t('site.signup.agree')"
                       >
+                        <!-- eslint-disable-next-line @intlify/vue-i18n/no-v-html -->
```
Comment (Mogge) [inline frontend/src/pages/Register.vue:103]:
```suggestion
```

**Review by Mogge — CHANGES_REQUESTED** [review]
Think of define the minus operator as well in the math key of the locales and use it inh amount mixin.

File: frontend/package.json [hunk 7]
```diff
@@ -8,11 +8,10 @@
     "build": "vue-cli-service build",
     "dev": "yarn run serve",
     "analyse-bundle": "yarn build && webpack-bundle-analyzer dist/webpack.stats.json",
-    "lint": "eslint --max-warnings=0 --ext .js,.vue .",
+    "lint": "eslint --max-warnings=0 --ext .js,.vue,.json .",
     "stylelint": "stylelint --max-warnings=0 '**/*.{scss,vue}'",
-    "i18n:report": "vue-cli-service i18n:report --src './src/**/*.?(js|vue)' --locales './src/locales/**/*.json'",
     "test": "TZ=UTC jest --coverage",
-    "locales": "scripts/missing-keys.sh && scripts/sort.sh"
+    "locales": "scripts/sort.sh"
```
Comment (Mogge) [inline frontend/package.json:14]: Why did you remove the missing keys here?

Comment (Mogge) [reply frontend/package.json:14]: This tests that the keys in de and en locales are exactly the same

**Review by Mogge — DISMISSED** [review]

File: frontend/public/json-example/admin_card_statistic.json [hunk 8]
```diff
@@ -1,36 +1,37 @@
-{"state":"success", "statisticdata": 
-	[
+{
```
[the remark this answers is not in the document]
Comment (ulfgebhardt) [reply frontend/public/json-example/admin_card_statistic.json:1]: I do not know - I just linted it - since it changed the line endings, I assume its data from Dario

File: frontend/src/components/GddSend/QrCode.vue [hunk 9]
```diff
@@ -1,5 +1,6 @@
 <template>
   <b-alert show variant="secondary">
+    <!-- eslint-disable-next-line @intlify/vue-i18n/no-v-html -->
     <span class="alert-text" v-html="$t('form.scann_code')"></span>
```
[the remark this answers is not in the document]
Comment (ulfgebhardt) [reply frontend/src/components/GddSend/QrCode.vue:4]: `"scann_code": "<strong>QR Code Scanner</strong> - Scan the QR Code of your partner",`
I am pretty sure this will not work, due to the html tag in the lang file. This is something the library prevents by default, which I disabled to not break the whole thing:
```
rules: {
        // TODO: enable
        '@intlify/vue-i18n/no-html-messages': 'off',
},
```

File: frontend/src/components/GddTransactionListFooter.vue [hunk 10]
```diff
@@ -2,10 +2,12 @@
   <div>
     <b-list-group>
       <b-list-group-item v-if="count > 5">
+        <!-- eslint-disable @intlify/vue-i18n/no-v-html -->
         <router-link
           to="/transactions"
-          v-html="$t('transaction.show_all', { count: count })"
+          v-html="$t('transaction.show_all', { count })"
         ></router-link>
```
[the remark this answers is not in the document]
Comment (ulfgebhardt) [reply frontend/src/components/GddTransactionListFooter.vue:9]: `"show_all": "View all <strong>{count}</strong> transactions."`

File: frontend/src/components/GddTransactionListFooter.vue [hunk 11]
```diff
@@ -2,10 +2,12 @@
   <div>
     <b-list-group>
       <b-list-group-item v-if="count > 5">
+        <!-- eslint-disable @intlify/vue-i18n/no-v-html -->
         <router-link
           to="/transactions"
-          v-html="$t('transaction.show_all', { count: count })"
+          v-html="$t('transaction.show_all', { count })"
```
[the remark this answers is not in the document]
Comment (ulfgebhardt) [reply frontend/src/components/GddTransactionListFooter.vue:8]: "show_all": "View all <strong>{count}</strong> transactions."

File: frontend/src/components/Menu/Navbar.spec.js [hunk 12]
```diff
@@ -54,31 +54,31 @@ describe('Navbar', () => {
...
       })
-      it('has first nav-item "transactions" in navbar', () => {
-        expect(wrapper.findAll('.nav-item').at(5).text()).toEqual('transactions')
+      it('has first nav-item "navigation.transactions" in navbar', () => {
+        expect(wrapper.findAll('.nav-item').at(5).text()).toEqual('navigation.transactions')
       })
-      it('has first nav-item "my-profil" in navbar', () => {
-        expect(wrapper.findAll('.nav-item').at(6).text()).toEqual('site.navbar.my-profil')
+      it('has first nav-item "navigation.my-profil" in navbar', () => {
+        expect(wrapper.findAll('.nav-item').at(6).text()).toEqual('navigation.my-profil')
```
[the remark this answers is not in the document]
Comment (ulfgebhardt) [reply frontend/src/components/Menu/Navbar.spec.js:67]: ???

File: frontend/package.json (hunk 7 above)
Comment (ulfgebhardt) [reply frontend/package.json:14]: I use `'@intlify/vue-i18n/no-missing-keys-in-other-locales': 'error',`

File: frontend/src/locales/de.json [hunk 13]
```diff
@@ -1,60 +1,51 @@
...
-      }
-    },
     "community": "Gemeinschaft",
     "continue-to-registration": "Weiter zur Registrierung",
     "current-community": "Aktuelle Gemeinschaft",
-    "location": "Ort:",
     "other-communities": "Weitere Gemeinschaften",
     "switch-to-this-community": "zu dieser Gemeinschaft wechseln"
   },
+  "dash": "—",
```
[the remark this answers is not in the document]
Comment (ulfgebhardt) [reply frontend/src/locales/de.json:12]: When I type the character in google I get results for `em-dash`

**Review by Mogge — APPROVED** [review]
